### PR TITLE
Gnomprona Monsters + Primal Ordeal Quest

### PR DESCRIPTION
**Pull request by carlospess0a**
# Note

I also fixing naga spells in this PR.

# Primal Ordeal Quest

- To be completed 

# Gnomprona monsters

## Bosses

- To be completed

### Sparkling Pools
image

- Emerald Tortoise
- Gore Horn
- Gorerilla
- Hulking Prehemoth
- Sabretooth

### Monster Graveyard [OK]
image

 - Nighthunter
 - Stalking Stalk
 - Sulphider
 - Sulphur Spouter
 - Undertaker
 
### Crystal Enigma [OK]
image

- Headpecker
- Mantosaurus
- Mercurial Menace
- Noxious Ripptor
- Shrieking Cry-stal

### RME Import monsters

> ```
> <monster name="Emerald Tortoise" looktype="1550" />
> <monster name="Headpecker" looktype="1557" />
> <monster name="Shrieking Cry-Stal" looktype="1560" />
> <monster name="Sulphur Spouter" looktype="1547" />
> <monster name="Hulking Prehemoth" looktype="1553" />
> <monster name="Gore Horn" looktype="1548" />
> <monster name="Gorerilla" looktype="1559" />
> <monster name="Nighthunter" looktype="1552" />
> <monster name="Sabretooth" looktype="1549" />
> <monster name="Stalking Stalk" looktype="1554" />
> <monster name="Mantosaurus" looktype="1556" />
> <monster name="Mercurial Menace" looktype="1561" />
> <monster name="Noxious Ripptor" looktype="1558" />
> <monster name="Sulphider" looktype="1546" />
> <monster name="Undertaker" looktype="1551" />
> ```

### CODE REVIEW

File: data/monster/amphibics/emeraldtortoise.lua [hunk 1]
```diff
@@ -0,0 +1,109 @@
+local mType = Game.createMonsterType("Emerald Tortoise")
+local monster = {}
+
+monster.name = "Emerald Tortoise"
```
Comment (guispiller) [inline data/monster/amphibics/emeraldtortoise.lua:4]: not used

File: data/monster/amphibics/emeraldtortoise.lua [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	lookType = 1550,
+	lookHead = 0,
+	lookBody = 0,
+	lookLegs = 0,
+	lookFeet = 0,
+	lookAddons = 0,
+	lookMount = 0
+}
+
+monster.raceId = 2268
```
Comment (guispiller) [inline data/monster/amphibics/emeraldtortoise.lua:17]: missing monster.Bestiary table

File: data/monster/amphibics/emeraldtortoise.lua [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+monster.health = 21040
+monster.maxHealth = 21040
+monster.runHealth = 0
```
Comment (guispiller) [inline data/monster/amphibics/emeraldtortoise.lua:20]: Our pattern uses this inside monster.flags

File: data/monster/amphibics/emeraldtortoise.lua [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+monster.race = "blood"
+monster.corpse = 39291
+monster.speed = 179
+monster.summonCost = 0
```
Comment (guispiller) [inline data/monster/amphibics/emeraldtortoise.lua:24]: it's called monster.manaCost

File: data/monster/amphibics/emeraldtortoise.lua [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+monster.changeTarget = {
+	interval = 2000,
+	chance = 0
+}
+
+monster.flags = {
```
Comment (guispiller) [inline data/monster/amphibics/emeraldtortoise.lua:31]: The pattern to all monsters is this order:  
 summonable =
	attackable = 
	hostile = 
	convinceable = 
	pushable =
	rewardBoss =
	illusionable = 
	canPushItems =
	canPushCreatures =
	staticAttackChance =
	targetDistance =
	runHealth = 
	healthHidden = 
	isBlockable = 
	canWalkOnEnergy = 
	canWalkOnFire = 
	canWalkOnPoison =

**Review by guispiller — CHANGES_REQUESTED** [review]
Great job, thanks for your contribution.

I'll just ask, please before taking it from draft, can you adept it to our pattern? All monsters use the same pattern/order of tags. I reviewed in one of the monsters, but it need to be applied in all of them

File: data/monster/amphibics/emeraldtortoise.lua (hunk 5 above)
Comment (carlospess0a) [reply data/monster/amphibics/emeraldtortoise.lua:31]: Thanks for the reply!

I will fix it.

**Comment by carlospess0a** [comment]
@guispiller hi, can you confirm if the Headpecker is in the pattern? I did the changes that you show me

**Comment by luiscaldera** [comment]
For those who want to add them to the rme, here I post it with the corresponding looktype, just add it to rowers-map-editor-12x\data\creatures.
monsters.xml

	<monster name="Emerald Tortoise" looktype="1550" />
	<monster name="Headpecker" looktype="1557" />
	<monster name="Shrieking Cry-Stal" looktype="1560" />
	<monster name="Sulphur Spouter" looktype="1547" />
	<monster name="Hulking Prehemoth" looktype="1553" />
	<monster name="Gore Horn" looktype="1548" />
	<monster name="Gorerilla" looktype="1559" />
	<monster name="Nighthunter" looktype="1552" />
	<monster name="Sabretooth" looktype="1549" />
	<monster name="Stalking Stalk" looktype="1554" />
	<monster name="Mantosaurus" looktype="1556" />
	<monster name="Mercurial Menace" looktype="1561" />
	<monster name="Noxious Ripptor" looktype="1558" />
	<monster name="Sulphider" looktype="1546" />
	<monster name="Undertaker" looktype="1551" />

**Comment by carlospess0a** [comment]
> For those who want to add them to the rme, here I post it with the corresponding looktype, just add it to rowers-map-editor-12x\data\creatures. monsters.xml
> 
> ```
> <monster name="Emerald Tortoise" looktype="1550" />
> <monster name="Headpecker" looktype="1557" />
> <monster name="Shrieking Cry-Stal" looktype="1560" />
> <monster name="Sulphur Spouter" looktype="1547" />
> <monster name="Hulking Prehemoth" looktype="1553" />
> <monster name="Gore Horn" looktype="1548" />
> <monster name="Gorerilla" looktype="1559" />
> <monster name="Nighthunter" looktype="1552" />
> <monster name="Sabretooth" looktype="1549" />
> <monster name="Stalking Stalk" looktype="1554" />
> <monster name="Mantosaurus" looktype="1556" />
> <monster name="Mercurial Menace" looktype="1561" />
> <monster name="Noxious Ripptor" looktype="1558" />
> <monster name="Sulphider" looktype="1546" />
> <monster name="Undertaker" looktype="1551" />
> 
> ```

Thanks for your contribution!

**Comment by majestyotbr** [comment]
The global datapack is migrating to Canary repository, you will need to create a new pull request there when the migration is ready.

**Pull request closed without merging**